echopop/survey.py

```diff
@@ -44,15 +44,38 @@ class Survey:
     """
 
     def __init__(
-        self, init_config_path: Union[str, Path], survey_year_config_path: Union[str, Path]
+        self,
+        init_config_path: Union[str, Path],
+        survey_year_config_path: Union[str, Path],
+        construct_nasc: bool = False,
+        transect_pattern: Optional[str] = r"T(\d+)",
+        index_variable: Union[str, List[str]] = ["transect_num", "interval"],
+        unique_region_id: str = "region_id",
+        region_class_column: str = "region_class",
+        export_file_directory: Optional[Union[str, Path]] = None,
+        export_save_directory: Optional[Union[str, Path]] = None,
```

Comment:
This would require refactoring (hopefully not too much!?), but seeing the large number of input arguments you added here, I think it would be better/cleaner if we keep the initialization of a Survey object just as initialization, and have the NASC ingestion part in a method. These arguments are really only used for the ingestion and not anywhere else. Separating it out as a method makes it explicit to the users what they are expected to do - to ingest NASC data (ie. not hiding everything under the hood of initialization, especially because it would take some time to compute), and also provide flexibility to skip the ingestion/load previously organized data.

Reply:
This would require either then moving `load_survey_data` out as well as a completely separate method (i.e. `Self.load_survey_data`) outside of the initialization, or shoehorning it into `Survey.transect_analysis`. This is because components of `load_survey_data` are contingent upon output files (i.e. the haul-transect key file) generated by the `batch_read_echoview_exports` function. 

> self.analysis["settings"].update(...) can be done in the method too.

I am a little unclear on what you are referring to here.

Reply:
Per discussion:
* Separate method for `batch_read_echoview_exports`
* Refactor `el.load_survey_data` as a separate method